vignettes/col_counts.Rmd

```diff
@@ -0,0 +1,130 @@
+
+```{r}
+library(dplyr)
+example_data <- data.frame(
+  ARMCD = c("ARM A", "ARM A", "ARM A", "ARM A", "ARM B", "ARM B"),
+  STRATA2 = c("S1", "S2", "S1", "S2",  "S2", "S2"),
+  STRATA1 = c("A", "A", "B", "B", "A", "B"),
+  AGE = c(20, 30, 40, 50, 21, 24)
+)
+
+lyt <- basic_table() %>%
+  split_cols_by("ARMCD", show_colcounts = TRUE, colcount_format = "N=xx") %>%
+  split_cols_by("STRATA2", show_colcounts = TRUE) %>%
+  split_cols_by("STRATA1") %>%
+  add_overall_col("All") %>%
+  analyze("AGE", afun = max, format = "xx.x")
+
+tbl <- build_table(lyt, example_data)
+tbl
+```
+
+We can control the column counts with the following functions:
+
+```{r}
+col_counts(tbl) # vector of all of them (visible and not visible) -> difficult to read/use for nested structures
+coldf <- make_col_df(tbl) # to check the structure of the columns as a data.frame
+coltree_structure(tbl) # Prints the structure of the columns as a tree
+```
+
+`make_col_df` is a fundamental function that is used also internally to keep track of the tree structure of the columns. It returns a `data.frame` very similar to `make_row_df`, but for columns. The `coltree_structure` function is a wrapper around `make_col_df` that prints the structure of the columns as a tree.
+
+To retrieve the column counts for a specific column, use pathing! Also visibility can be accessed and modified with the following functions:
+
+```{r}
+facet_colcount(tbl, coldf$path[[1]]) # To get the column counts for a specific column at a specific level
+colcount_visible(tbl, coldf$path[[1]]) # not visible!!! # FALSE
+facet_colcount(tbl, coldf$path[[1]][c(1, 2)]) # visible
+colcount_visible(tbl, coldf$path[[1]][c(1, 2)]) # TRUE
+```
+
+Formats can differ as we have shown in the example above. We can change the format of the column counts with the following function:
+
+```{r, eval = FALSE}
+# Modifying last level!?
+colcount_format(tbl) <- "[N<=xx]" # Does not work
+colcount_visible(tbl, coldf$path[[1]]) <- TRUE # not the right way!!
+# > tbl # ERROR! we need to set the other values too
+# Error in h(simpleError(msg, call)) :
+#   error in evaluating the argument 'x' in selecting a method for function 'toString':
+#   Detected different colcount visibility among sibling facets (those arising from the
+#   same split_cols_by* layout instruction). This is not supported.
+# Set count values to NA if you want a blank space to appear as the displayed count for particular facets.
+# First disagreement occured at paths:
+# ARMCD[ARM A]->STRATA2[S1]->STRATA1[A]
+# ARMCD[ARM A]->STRATA2[S1]->STRATA1[B]
+# colcount_visible(tbl, coldf$path[[1]]) <- FALSE
+
+# Trying with col_counts
+col_counts(tbl) <- c(1, rep(NA_integer_, length(col_counts(tbl)) - 1)
+# colcount_visible(tbl, coldf$path[[1]]) <- TRUE # same error as above we need facets!!
+facet_colcounts_visible(tbl, c("ARMCD", "*", "STRATA2", "*", "STRATA1")) <- TRUE # does not work
+print(tbl)
+```
```

Comment:
@gmbecker I tried here to change the last facet only. So changing format and visibility. It seems that facet_colcounts_visible does not work on the last facet (I think it is bc here there is All too). Tried different paths but all failing to me

Reply:
@Melkiades I took a pass at writing the colcounts vignette. Take a look at what I pushed and see if it works for you, if so I think we can drop most of what was in there

Reply:
Thank you!! it is much more intuitive now, removed the other part of the vignette